server/controllers/article.controller.js

```diff
@@ -20,25 +20,123 @@ class ArticleController {
     */
     if (req.file) req.body.image = req.file.secure_url;
     const {
-      title, body, tags, description, image, published
+      title, body, tagsList, description, published
     } = req.body;
     const { id } = req.user;
     const successMessage = 'Article successfully created';
     const errorMessage = 'Article was not created successfully';
+    const tags = [...new Set(tagsList)];
     const article = await Article.create({
       title,
       body,
-      tags,
+      tagsList,
       description,
-      image,
+      image: req.body.image,
       published,
-      slug: slug(`${title} ${Date.now()}`),
+      slug: slugify(`${title} ${Date.now()}`),
       userId: id
     });
+    if (tags) {
+      let insertTags = tags.map(async (newTags) => {
+        const newtag = await tag.findOrCreate({ where: { name: newTags } });
+        return newtag[0];
+      });
+
+      insertTags = await Promise.all(insertTags);
```

Comment:
Hi @davidokonji  A little explanation on the `Promise.all` please.

Reply:
@Chiazokam the `Promise.all` is used to resolve the promise which was sent from the `insertTags` array which returns a promise

Reply:
Alright..thanks